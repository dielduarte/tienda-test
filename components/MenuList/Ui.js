import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Small } from '../Text';

export const Menu = styled.nav`
  list-style-type: none;
`;

export const MenuItemText = styled(Small)`
  margin-left: 10px;
  flex: 2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const MenuIcon = styled(FontAwesomeIcon)`
  margin-right: 1px;
`;

export const MenuItem = styled.div`
  padding: 15px 10px;
  display: flex;
  align-items: center;
  color: ${props => props.theme.white};
  transition: all 0.2s linear;

  &:hover {
    cursor: pointer;
    color: ${props => props.theme.lightBlue};
  }
`;
