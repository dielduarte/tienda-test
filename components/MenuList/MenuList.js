import React from 'react';
import Link from 'next/link';
import { arrayOf, objectOf, string } from 'prop-types';

import { Row } from '../Flex';
import { Menu, MenuItem, MenuItemText, MenuIcon } from './Ui';

function renderItem(item, key) {
  return (
    <Link href={item.href} key={key}>
      <MenuItem>
        <Row>
          <MenuIcon icon="ellipsis-v" />
          <MenuIcon icon="bars" />
        </Row>
        <MenuItemText>{item.title}</MenuItemText>
        <MenuIcon icon="angle-right" size="lg" />
      </MenuItem>
    </Link>
  );
}

function MenuList({ items }) {
  return <Menu>{items.map((item, key) => renderItem(item, key))}</Menu>;
}

MenuList.propsType = {
  items: arrayOf(objectOf({ title: string, href: string }))
};

export default MenuList;
