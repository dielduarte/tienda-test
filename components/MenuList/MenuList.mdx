---
name: MenuList
---

import '../../config/fontAwesome';
import { Playground, PropsTable } from 'docz'
import MenuList from './MenuList';

# MenuList

<Playground style={{ background: '#333' }}>
  <MenuList items={[
      { title: 'Produtos', href: '/products' },
      { title: 'Adicionar um novo produto', href: '/products/new' }
    ]}
  />
</Playground>