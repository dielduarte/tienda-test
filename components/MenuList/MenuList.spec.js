import { shallow } from 'enzyme';
import 'jest-styled-components';

import MenuList from './MenuList';

describe('MenuList', () => {
  it('should render correcly', () => {
    const wrapper = shallow(
      <MenuList
        items={[
          { title: 'Produtos', href: '/products' },
          { title: 'Adicionar um novo produto', href: '/products/new' }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
