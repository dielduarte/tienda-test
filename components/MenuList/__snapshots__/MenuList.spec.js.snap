// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList should render correcly 1`] = `
<styled.nav>
  <Link
    href="/products"
    key="0"
  >
    <styled.div>
      <styled.div>
        <Styled(FontAwesomeIcon)
          icon="ellipsis-v"
        />
        <Styled(FontAwesomeIcon)
          icon="bars"
        />
      </styled.div>
      <Styled(styled.h3)>
        Produtos
      </Styled(styled.h3)>
      <Styled(FontAwesomeIcon)
        icon="angle-right"
        size="lg"
      />
    </styled.div>
  </Link>
  <Link
    href="/products/new"
    key="1"
  >
    <styled.div>
      <styled.div>
        <Styled(FontAwesomeIcon)
          icon="ellipsis-v"
        />
        <Styled(FontAwesomeIcon)
          icon="bars"
        />
      </styled.div>
      <Styled(styled.h3)>
        Adicionar um novo produto
      </Styled(styled.h3)>
      <Styled(FontAwesomeIcon)
        icon="angle-right"
        size="lg"
      />
    </styled.div>
  </Link>
</styled.nav>
`;
